probe-rs/src/flashing/loader.rs

```diff
@@ -12,61 +12,46 @@ use std::{
     io::{Read, Seek, SeekFrom},
 };
 
-struct RamWrite<'data> {
+struct RamWrite {
     address: u32,
-    data: &'data [u8],
+    data: Vec<u8>,
 }
 
 /// `FlashLoader` is a struct which manages the flashing of any chunks of data onto any sections of flash.
 /// Use `add_data()` to add a chunks of data.
 /// Once you are done adding all your data, use `commit()` to flash the data.
 /// The flash loader will make sure to select the appropriate flash region for the right data chunks.
 /// Region crossing data chunks are allowed as long as the regions are contiguous.
-pub struct FlashLoader<'data> {
+pub struct FlashLoader {
     memory_map: Vec<MemoryRegion>,
-    builders: HashMap<NvmRegion, FlashBuilder<'data>>,
-    ram_write: Vec<RamWrite<'data>>,
-    keep_unwritten: bool,
+    builders: HashMap<NvmRegion, FlashBuilder>,
+    ram_write: Vec<RamWrite>,
 
     /// Source of the flash description,
     /// used for diagnostics.
     source: TargetDescriptionSource,
 }
 
-impl<'data> FlashLoader<'data> {
+impl FlashLoader {
     /// Create a new flash loader.
-    pub fn new(
-        memory_map: Vec<MemoryRegion>,
-        keep_unwritten: bool,
-        source: TargetDescriptionSource,
-    ) -> Self {
+    pub fn new(memory_map: Vec<MemoryRegion>, source: TargetDescriptionSource) -> Self {
         Self {
             memory_map,
             builders: HashMap::new(),
             ram_write: Vec::new(),
-            keep_unwritten,
             source,
         }
     }
 
-    pub(super) fn add_section(
```

Comment:
Please keep this function, it works differently than the `add_data` function. The information in `ExtractedFlashData` is lost if only the `add_data` function is used.

Reply:
Where is that information being used? The previous `add_data_internal` code only accessed `data.address()` and `data.data()`, so the extra info was already being thrown away. Seeing that, I took the opportunity to simplify. Is that not the case?

Reply:
Hmm, you're right. I think I planned to use it, and haven't actually gotten round to do so.